internal/util/cache/lru_cache.go

```diff
@@ -90,35 +114,38 @@ func (c *LRU) evictedWorker() {
 			return
 		case e, ok := <-c.evictedCh:
 			if ok {
-				if c.onEvicted != nil {
-					c.onEvicted(e.key, e.value)
+				if c.storeHelper.OnEvicted != nil {
+					c.storeHelper.OnEvicted(e.Key, e.Value)
 				}
 			}
 		}
 	}
 }
 
-func (c *LRU) Add(key, value Value) {
+func (c *LRU) Add(key Key, value Value) {
 	c.m.Lock()
 	defer c.m.Unlock()
 	c.stats.writeCount++
 	if e, ok := c.items[key]; ok {
 		c.evictList.MoveToFront(e)
-		e.Value.(*entry).value = value
+		e.Value.(*Entry).Value = value
 		return
 	}
-	e := &entry{key: key, value: value}
+	e := c.storeHelper.Store(key, value)
+	e.Size = c.storeHelper.MeasureSize(value)
 	listE := c.evictList.PushFront(e)
 	c.items[key] = listE
+	c.size += uint64(e.Size)
 
-	if c.evictList.Len() > c.capacity {
+	for c.size > c.capacity {
 		c.stats.evictedCount++
 		oldestE := c.evictList.Back()
 		if oldestE != nil {
 			c.evictList.Remove(oldestE)
-			kv := oldestE.Value.(*entry)
-			delete(c.items, kv.key)
-			if c.onEvicted != nil {
+			kv := oldestE.Value.(*Entry)
+			c.size -= uint64(kv.Size)
```

Comment:
why not use  c.storeHelper.MeasureSize?

Reply:
For measureSize may produce overhead, I hope it is used for once. So size has been assign into Entry in line 135.